lib/common/util.js

```diff
@@ -452,102 +451,228 @@ function shouldRetry(err) {
+    if (err) {
+      // google-auth-library returns a "Could not load..." error if it can't get
+      // an access token. However, it's possible an API request doesn't need to
+      // be authenticated, e.g. when downloading a file from a public bucket. We
+      // consider this error a warning, and allow the request to go through
+      // without authorization, relying on the upstream API to return an error
+      // the user would find more helpful, should one occur.
+      if (err.message.indexOf('Could not load') === 0) {
+        callback(null, reqOpts);
+      } else {
+        callback(err);
+      }
+      return;
+    }
+
+    authClient.getAccessToken(function(err, token) {
       if (err) {
-        if (GAE_OR_GCE && err.code === 'ENOTFOUND') {
-          // The metadata server wasn't found. The user must not actually be in
-          // a GAE or GCE environment.
-          throw missingCredentialsError;
-        }
+        callback(err);
+        return;
+      }
 
-        if (err.code === 401 &&
-            ++tokenRefreshAttempts <= MAX_TOKEN_REFRESH_ATTEMPTS) {
-          authorize(reqOpts, onAuthorizedRequest);
-          return;
+      var authorizedReqOpts = extend(true, {}, reqOpts, {
+        headers: {
+          Authorization: 'Bearer ' + token
         }
+      });
 
-        // For detecting Sign errors on io.js (1.x) (or node 0.11.x)
-        // E.g. errors in form: error:code:PEM routines:PEM_read_bio:error_name
-        var pemError = err.message &&
-            err.message.indexOf('error:') !== -1;
+      callback(null, authorizedReqOpts);
+    });
+  });
+}
 
-        if (err.message === 'SignFinal error' || pemError) {
-          err.message = [
-            'Your private key is in an unexpected format and cannot be used.',
-            'Please try again with another private key.'
-          ].join(' ');
-        }
+util.authorizeRequest = authorizeRequest;
 
-        (callback.onAuthorized || callback)(err);
-        return;
+/**
+ * Get a function for making authorized requests.
+ *
+ * @param {object} config - Configuration object.
+ * @param {boolean=} config.autoRetry - Automatically retry requests if the
+ *     response is related to rate limits or certain intermittent server errors.
+ *     We will exponentially backoff subsequent requests by default. (default:
+ *     true)
+ * @param {object=} config.credentials - Credentials object.
+ * @param {boolean=} config.customEndpoint - If true, just return the provided
+ *     request options. Default: false.
+ * @param {string=} config.email - Account email address, required for PEM/P12
+ *     usage.
+ * @param {number=} config.maxRetries - Maximum number of automatic retries
+ *     attempted before returning the error. (default: 3)
+ * @param {string=} config.keyFile - Path to a .json, .pem, or .p12 keyfile.
+ * @param {array} config.scopes - Array of scopes required for the API.
+ */
+function makeAuthorizedRequestFactory(config) {
```

Comment:
`makeAuthorizedRequest` -> `makeAuthorizedRequestFactory` since that's a little more telling of how it works. Also documented the returned function (`makeAuthorizedRequest`).